src/System.Windows.Forms/src/System/Windows/Forms/TreeView.TreeViewAccessibleObject.cs

```diff
@@ -0,0 +1,136 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Drawing;
+using static Interop;
+using static System.Windows.Forms.TreeNode;
+
+namespace System.Windows.Forms
+{
+    public partial class TreeView
+    {
+        internal class TreeViewAccessibleObject : ControlAccessibleObject
+        {
+            private readonly TreeView _owningTreeView;
+
+            public TreeViewAccessibleObject(TreeView owningTreeView) : base(owningTreeView)
+            {
+                _owningTreeView = owningTreeView ?? throw new ArgumentNullException(nameof(owningTreeView));
+            }
+
+            internal override UiaCore.IRawElementProviderFragmentRoot FragmentRoot => this;
+
+            internal override UiaCore.IRawElementProviderFragment? FragmentNavigate(UiaCore.NavigateDirection direction)
+                => direction switch
+                {
+                    UiaCore.NavigateDirection.FirstChild => GetChild(0),
+                    UiaCore.NavigateDirection.LastChild => GetChild(GetChildCount()-1),
+                    _ => base.FragmentNavigate(direction),
+                };
+
+            public override AccessibleObject? GetChild(int index)
+                => index >= 0 && index < GetChildCount()
+                    ? _owningTreeView.Nodes[index].AccessibilityObject
+                    : null;
+
+            public override int GetChildCount() => _owningTreeView.Nodes.Count;
+
+            internal override int GetChildIndex(AccessibleObject? child)
+                => child is not null && child is TreeNodeAccessibleObject node ? node.Index : -1;
```

Comment:
`child is TreeNodeAccessibleObject node` implies that `child is not null`:
```suggestion
                => child is TreeNodeAccessibleObject node ? node.Index : -1;
```